compiler/lib/driver.ml

```diff
@@ -286,21 +286,29 @@ let link ~standalone ~linkall ~export_runtime (js : Javascript.source_elements)
 
 let macro recurse fallthrough =
   let module J = Javascript in
+  let zero, one = J.ENum "0", J.ENum "1" in
   function
+  | "BLOCK", [_] ->
+    failwith (Format.sprintf "macro BLOCK called with inappropriate arguments")
   | "BLOCK", tag :: args ->
```

Comment:
You can use
`"BLOCK", tag :: ( _ :: _ as args)`
and get rid of the match on "BLOCK" above.